src/object/cli_obj.c

```diff
@@ -1409,6 +1411,25 @@ obj_shards_2_fwtgts(struct dc_object *obj, uint32_t map_ver, uint8_t *bit_map,
 		return rc;
 	}
 
+	if (obj_auxi->spec_shard) {
+		D_ASSERT(grp_nr == 1);
+		D_ASSERT(shard_cnt == 1);
+		D_ASSERT(bit_map == NIL_BITMAP);
+		D_ASSERT(req_tgts->ort_srv_disp == 0);
+
+		tgt = req_tgts->ort_shard_tgts;
+
+		req_tgts->ort_grp_nr = 1;
+		req_tgts->ort_grp_size = 1;
+		if (obj_is_ec(obj))
+			req_tgts->ort_start_shard = rounddown(start_shard, obj_get_grp_size(obj));
```

Comment:
(start_shard / obj_get_grp_size(obj)) * daos_oclass_grp_size(oca);?

Reply:
oh, 2.2 does not support online osa anyway, so daos_oclass_grp_size == obj_get_grp_size() anyway. Though I will update the patch if it needs to.